Cache/IO.lean

```diff
@@ -92,21 +93,44 @@ abbrev PackageDirs := Lean.RBMap String FilePath compare
 def isMathlibRoot : IO Bool :=
   FilePath.mk "Mathlib" |>.pathExists
 
-def mathlibDepPath : FilePath :=
-  LAKEPACKAGESDIR / "mathlib"
+def parseMathlibDepPath (json : Lean.Json) : Except String (Option FilePath) := do
+  let deps ← (← json.getObjVal? "packages").getArr?
+  for d in deps do
+    let n := ← (← d.getObjVal? "name").getStr?
+    if n != "mathlib" then
+      continue
+    let t := ← (← d.getObjVal? "type").getStr?
+    if t == "path" then
+      return some ⟨← (← d.getObjVal? "dir").getStr?⟩
+    else
+      return LAKEPACKAGESDIR / "mathlib"
+  return none
+
+def mathlibDepPath : IO FilePath := do
+  let raw ← IO.FS.readFile "lake-manifest.json"
```

Comment:
Add a `try / catch` here, and a helpful error message about what to do about a missing `lake-manifest.json`.

Perhaps:
"If this is a new project, you will need to run `lake build` first in order for construct your initial `lake-manifest.json`. If this is an existing project, contact the authors to ask that they include the `lake-manifest.json` file."

Reply:
(We just had someone on zulip asking if they should .gitignore their lake-manifest.json... So people are going to do it!)

Reply:
The act of running `lake exe cache get` appears to create the manifest file; so I think this is basically unreachable, and the help text you suggest is unlikely to help anyone reaching it.

Reply:
I confirmed that `lake exe nonexistant` also creates `lake-manifest.json`, even if we disable the post-update hook in Mathlib; so I think this is not anything to worry about.

(Weirdly, if the manifest is missing, `lake exe nonexistant` also runs `lake exe get cache`, because of the post-update hook!)

Reply:
Great!